engine/src/main/java/org/terasology/rendering/backdrop/Skysphere.java

```diff
@@ -101,16 +101,30 @@ public float getSunPositionAngle() {
         return celSystem.getSunPosAngle();
     }
 
+    /**
+     * Get the strength of the light for the current day part, based on sun/moon visibility.
+     * @return the strength of the current light in [0;1] range.
+     */
     @Override
     public float getDaylight() {
-        float angle = (float) Math.toDegrees(TeraMath.clamp(Math.cos(getSunPositionAngle())));
-        float daylight = 1.0f;
+        float lightBasedOnTimeOfDay = 1.0f;
+        final float ambientLight = 0.2f; // not totally black when the sun goes down, but the moon is not yet up
+        double radSunAngle = getSunPositionAngle(); // expected [-PI;PI] for the day cycle to work correctly!
+
+        // if the degree of the sun/moon is less than "lessLightUnderAngle", then we gradually increase the light, instead of full intensity
+        final double lessLightUnderAngle = Math.sin(Math.toRadians(15.0));
```

Comment:
@Avalancs: I don't know if an asymmetrical ramping up/down of intensity is going to make a significant difference. If I remember correctly MC has a 7 minutes night and a 13 minutes day, but I don't remember a defined transition between them and the light intensity never really goes down too much. In Terasology at this stage we can do whatever we want and it would be interesting from a gameplay perspective to have at least a few minutes of complete darkness. It's an easy way to be different from MC.

@skaldarnar: My apologies good Sir. I shall go to my cell and repent by reading for 100 times the CheckStyle rules files out loud.